src/core/views.py

```diff
@@ -2262,38 +2266,46 @@ def get_facet_queryset(self):
         # instead of a separate facet.
         # return None
         queryset = self.filter_queryset_if_journal(
-            self.model.objects.all()
+            super().get_queryset()
         ).exclude(
             stage=submission_models.STAGE_UNSUBMITTED
         )
         facets = self.get_facets()
         for facet in facets.values():
             queryset = queryset.annotate(**facet.get('annotations', {}))
-        return queryset
+        return queryset.order_by()
 
     def get_actions(self):
         return []
 
     def post(self, request, *args, **kwargs):
 
-        action_status = ''
-        action_error = False
         params_string = request.POST.get('params_string')
         params_querydict = QueryDict(params_string, mutable=True)
+
         actions = self.get_actions()
-        queryset = self.get_queryset(params_querydict=params_querydict)
-        if request.journal:
-            querysets = [queryset]
-        else:
+        if actions:
+            start = datetime.datetime.now()
+
+            action_status = ''
+            action_error = False
+
             querysets = []
-            for journal in {article.journal for article in queryset}:
-                querysets.append(queryset.filter(journal=journal))
+            queryset = self.get_queryset(params_querydict=params_querydict)
```

Comment:
We are not sanitising `params_string`, meaning that users would be able to inject their own database filters here with `params_querydict` outside of those you have intended.

We should make sure the user has not sent additional 'facets' than those we expect them to. This can be done in `.get_queryset` but also validating `forms.CBVFacetForm` with `request.POST`/ `request.GET` data as appropriate

Reply:
Yes, this occurred to me, and I made sure to filter out unexpected parameters in `get_queryset` with this line: https://github.com/BirkbeckCTP/janeway/blob/d6718e24980d3b4d5d448fb337dc1ddb57041231/src/core/views.py#L2233

I'll add a comment above it specifying that's why it's there. Let me know if you spot a problem with it.